calico-vpp-agent/cni/network_vpp.go

```diff
@@ -30,6 +30,18 @@ func getInterfaceVrfName(podSpec *storage.LocalPodSpec, suffix string) string {
 	return fmt.Sprintf("pod-%s-table-%s", podSpec.Key(), suffix)
 }
 
+func (s *Server) checkAvailableBuffers() (uint64, error) {
+	NumRxQueues := config.TapNumRxQueues
+	NumTxQueues := config.TapNumTxQueues
+	RxQueueSize := vpplink.DefaultIntTo(config.TapRxQueueSize, vpplink.DEFAULT_QUEUE_SIZE)
+	TxQueueSize := vpplink.DefaultIntTo(config.TapTxQueueSize, vpplink.DEFAULT_QUEUE_SIZE)
+	buffersNeeded := uint64(RxQueueSize * NumRxQueues + TxQueueSize * NumTxQueues)
+	if buffersNeeded > s.availableBuffers {
+		return buffersNeeded, errors.Errorf("Cannot create interface: Out of buffers: available buffers = %d, buffers needed = %d", s.availableBuffers, buffersNeeded)
```

Comment:
It's great to indicate the values in the message, to make it even better we could tell the user how to increase the number of buffers and how to prevent the error by appending something like: `Increase buffers-per-numa in the VPP configuration or reduce CALICOVPP_TAP_RING_SIZE the to allow more pods to be scheduled. You may want to limit the number of pods per node to prevent this error`.